### PR TITLE
Add gradle build scripts for test codes and improve travis scripts

### PR DESCRIPTION
**Pull request by daisy-ycguo**
In this patch, I added gradle build scripts for openwhisk-package-template. Now it's able to run the test codes under openwhisk-package-template by:
step 1. set a system environment variable "OPENWHISK_HOME" to the folder where openwhisk is cloned.
step 2. run "./gradlew :tests:test"

I also improved travis scripts. In travis, openwhisk is cloned and installed. Then openwhisk-package-template is installed. And then, test codes are invoked.

I have tested the travis in my forked repo. It works fine now.

### CODE REVIEW

**Comment by csantanapr** [comment]
@daisy-ycguo please assigned PRs to Vincent for this repo. 
I'm out of the office for a few weeks

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,8 @@
+<<<<<<< HEAD
```
Comment (rabbah) [inline .gitignore]: Bad merge?

File: tests/build.gradle [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    ':core:nodejsAction:distDocker',
+    ':core:nodejs6Action:distDocker',
+    ':core:pythonAction:distDocker',
+    ':core:javaAction:distDocker',
+    ':core:swiftAction:distDocker',
+    ':core:swift3Action:distDocker'
+])*/
+
+dependencies {
+    testCompile 'org.scala-lang:scala-library:2.11.7'
```
Comment (rabbah) [inline tests/build.gradle:43]: Are you sure you need all these dependencies to run package tests?

File: tests/build.gradle [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                      '$whiskhome/core/controller/build/libs/controller.jar', 
+                      '$whiskhome/core/dispatcher/build/libs/dispatcher.jar')
+    testCompile files(test_common)
+}
+
+def keystorePath = new File(buildDir, 'classes/test/keystore')
+task deleteKeystore(type: Delete) {
+    delete keystorePath
+}
+task createKeystore(dependsOn: deleteKeystore) << {
```
Comment (rabbah) [inline tests/build.gradle]: You don't need these tasks either

File: tests/build.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<!-- vim: set expandtab ts=4 sw=4 : -->
+<project basedir="..">
```
Comment (rabbah) [inline tests/build.xml]: Suggest you don't bother copying any of the ant stuff. We are removing ant.

**Comment by csantanapr** [comment]
Actually assigned to Rodric since his reviewing this PRs
Thanks PRs I agree let's not include stuff we really don't need

**Comment by daisy-ycguo** [comment]
@rabbah  Thank you to help me clarify these tasks. I don't clearly understand what they are doing.
I will upload a new version next Monday.

**Pull request closed without merging**